### PR TITLE
A Rails form web page created through with various methods

### PR DESCRIPTION
**Pull request by ikraamg**
Good day,

Please find the Readme [here](https://github.com/ikraamg/re-former/blob/form/README.md)

In this project, we:
- create the necessary MVC elements to build a user form that can be created, viewed and edited.
- In addition, if a form fails to save or update an entry, then a list of errors will be displayed in the view.
- Added validation on input
- Used gitflow.
- Rubocop for linting errors.

Thanks!

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  resources :users
```
Comment (shshamim63) [inline config/routes.rb]: - Use the :only option to specify just the :new, :create, ;edit and :update actions in config/routes.rb ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_route2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<% if @user %>
+<label>Username:</label> <%= @user.username %> <br />
+<label>Email:</label> <%= @user.email %>
+<%end%>
```
Comment (shshamim63) [inline app/views/users/show.html.erb]: Fix the indentation.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :point_right: Required Changes :point_left:
___
Hi @ikraamg and @alishabab :wave:
Awesome initiative, the project contains a descriptive readme. You almost completed the project. I just found an issue and that is blocking me to approve it. Let's get to it.
___
### :spades: Mandatory Changes :spades:
- You have to use the only: on your route. Please check the comment below on your routes.rb file.
- Please fix the problem related to indentation.
___
#### Please fix the above-mentioned issue and send another review request.
___
## :herb: Happy Coding :herb:
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      redirect_to user_path(@user)
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb:14]: You can  use `redirect_to @user`

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    if @user.update(user_params)
+      redirect_to user_path
```
Comment (kobiyoyo) [inline app/controllers/users_controller.rb:28]: You can  use `redirect_to @user`

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  <input type="password" name="user[password]" id="password"><br>
+  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>
+<%# <button type="submit">submit</button> %>
+
+
+<%# </form> %>
+
+<% if false %>
+
+<%= form_tag("/users") do %>
```
Comment (kobiyoyo) [inline app/views/users/new.html.erb:17]: lets write it this way `<%= form_tag(users_path) do %>`

**Review by kobiyoyo — APPROVED** [review]
# Project Approved :thumbsup:
## Hello Team,  You have done a great job with this project. :clap: ,just few changes required
- [ ] check comments below
#### You can go ahead and merge once done!!!

**Pull request merged**